TVShows/TVShows/Modules/AddNewEpisode/AddNewEpisodeWireframe.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  AddNewEpisodeWireframe.swift
+//  TVShows
+//
+//  Created by Ivan Milicevic on 22/10/2018.
+//  Copyright (c) 2018 Ivan Milicevic. All rights reserved.
+//
+//  This file was generated by the 🐍 VIPER generator
+//
+
+import UIKit
+
+final class AddNewEpisodeWireframe: BaseWireframe {
+
+    // MARK: - Private properties -
+
+    private let _storyboard = UIStoryboard(name: "AddNewEpisode", bundle: nil)
+    private weak var _delegate: ShowDetailsPresenterInterface?
+
+    // MARK: - Module setup -
+
+    init(show: Show, delegate: ShowDetailsPresenterInterface) {
+        let moduleViewController = _storyboard.instantiateViewController(ofType: AddNewEpisodeViewController.self)
+        super.init(viewController: moduleViewController)
+        
+        let interactor = AddNewEpisodeInteractor()
+        let presenter = AddNewEpisodePresenter(wireframe: self, view: moduleViewController, interactor: interactor, show: show)
+        moduleViewController.presenter = presenter
+        _delegate = delegate
+    }
+
+
+}
+
+// MARK: - Extensions -
+
+extension AddNewEpisodeWireframe: AddNewEpisodeWireframeInterface {
+
+    func navigate(to option: AddNewEpisodeNavigationOption) {
+
+        switch option {
+        case .showDetails(let didAdd):
+            _returnToShowDetails(didAdd: didAdd)
+        }
+
+    }
+
+    private func _returnToShowDetails(didAdd: Bool) {
+
+        if didAdd {
+            guard let _delegate = _delegate else {
```

Comment:
No need for unwrapping delegate. Just use optional chaining:

```swift
if didAdd
    _delegate?.reloadData()
}
```

if delegate exists it will call reloadData, and if not it will just ignore call (because of usage of `?` chain operator)